packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -48,49 +49,80 @@ class MouseTrackerAnnotation {
 
   @override
   String toString() {
-    final String none = (onEnter == null && onExit == null && onHover == null) ? ' <none>' : '';
-    return '[$runtimeType${hashCode.toRadixString(16)}$none'
-        '${onEnter == null ? '' : ' onEnter'}'
-        '${onHover == null ? '' : ' onHover'}'
-        '${onExit == null ? '' : ' onExit'}]';
+    final List<String> callbacks = <String>[];
+    if (onEnter != null)
+      callbacks.add('enter');
+    if (onHover != null)
+      callbacks.add('hover');
+    if (onExit != null)
+      callbacks.add('exit');
+    final String describeCallbacks = callbacks.isEmpty
+      ? '<none>'
+      : callbacks.join(' ');
+    return '${describeIdentity(this)}(callbacks: $describeCallbacks)';
   }
 }
 
-// Used internally by the MouseTracker for accounting for which annotation is
-// active on which devices inside of the MouseTracker.
-class _TrackedAnnotation {
-  _TrackedAnnotation(this.annotation);
-
-  final MouseTrackerAnnotation annotation;
-
-  /// Tracks devices that are currently active for this annotation.
-  ///
-  /// If the mouse pointer corresponding to the integer device ID is
-  /// present in the Set, then it is currently inside of the annotated layer.
-  ///
-  /// This is used to detect layers that used to have the mouse pointer inside
-  /// them, but now no longer do (to facilitate exit notification).
-  Set<int> activeDevices = <int>{};
-}
-
-/// Describes a function that finds an annotation given an offset in logical
-/// coordinates.
+/// Signature for searching for [MouseTrackerAnnotation]s at the given offset.
 ///
 /// It is used by the [MouseTracker] to fetch annotations for the mouse
 /// position.
 typedef MouseDetectorAnnotationFinder = Iterable<MouseTrackerAnnotation> Function(Offset offset);
 
-/// Keeps state about which objects are interested in tracking mouse positions
-/// and notifies them when a mouse pointer enters, moves, or leaves an annotated
-/// region that they are interested in.
+// Various states of each connected mouse device.
+//
+// It is used by [MouseTracker] to compute which callbacks should be triggered
+// by each event.
+class _MouseState {
+  _MouseState({
+    @required PointerEvent mostRecentEvent,
+  }) : assert(mostRecentEvent != null),
+       _mostRecentEvent = mostRecentEvent;
+
+  // The list of annotations that contains this device during the last frame.
+  LinkedHashSet<MouseTrackerAnnotation> lastAnnotations =
```

Comment:
nit: Leave this on one line to improve readability (Flutter's style guide does not have a hard 80 char cutoff limit, you can go over to make things more readable)